Packages/ClientRuntime/Sources/PrimitiveTypeExtensions/String+Extensions.swift

```diff
@@ -124,3 +124,30 @@ public extension String {
         return self
     }
 }
+
+/// subscript string extensions
+public extension StringProtocol {
+    subscript(_ offset: Int) -> Element {
+        self[index(startIndex, offsetBy: offset)]
+    }
+    
+    subscript(_ range: Range<Int>) -> SubSequence {
+        prefix(range.lowerBound+range.count).suffix(range.count)
+    }
+    
+    subscript(_ range: ClosedRange<Int>) -> SubSequence {
+        prefix(range.lowerBound+range.count).suffix(range.count)
+    }
+    
+    subscript(_ range: PartialRangeThrough<Int>) -> SubSequence {
+        prefix(range.upperBound.advanced(by: 1))
+    }
+    
+    subscript(_ range: PartialRangeUpTo<Int>) -> SubSequence {
+        prefix(range.upperBound)
+    }
+    
+    subscript(_ range: PartialRangeFrom<Int>) -> SubSequence {
+        suffix(Swift.max(0, count-range.lowerBound))
+    }
```

Comment:
Aren't these native to Swift?

Reply:
Good call

There are [subscript implementation]( https://developer.apple.com/documentation/swift/string/subscript(_:)-lc0v) for string but they are `Index` based which are almost impossible to use in real life.